addons/stock/tests/test_old_rules.py

```diff
@@ -593,16 +593,40 @@ def test_pick_ship_1(self):
             'warehouse_id': self.warehouse_2_steps.id,
             'group_id': procurement_group2.id,
             'origin': 'origin2',
-        })
+        },
+        {
+            'name': 'third out move',
+            'procure_method': 'make_to_order',
+            'location_id': ship_location.id,
+            'location_dest_id': customer_location.id,
+            'product_id': self.productB.id,
+            'product_uom': self.uom_unit.id,
+            'product_uom_qty': 1.0,
+            'warehouse_id': self.warehouse_2_steps.id,
+            'group_id': procurement_group2.id,
+            'origin': 'origin2',
+        },
+        {
+            'name': 'fourth out move',
+            'procure_method': 'make_to_order',
+            'location_id': ship_location.id,
+            'location_dest_id': customer_location.id,
+            'product_id': self.productB.id,
+            'product_uom': self.uom_unit.id,
+            'product_uom_qty': 1.0,
+            'warehouse_id': self.warehouse_2_steps.id,
+            'group_id': procurement_group1.id,
+            'origin': 'origin1',
+        }])
 
         # first out move, the "pick" picking should have a partner and an origin
         move1._action_confirm()
         picking_pick = move1.move_orig_ids.picking_id
         self.assertEqual(picking_pick.partner_id.id, procurement_group1.partner_id.id)
         self.assertEqual(picking_pick.origin, move1.group_id.name)
 
-        # second out move, the "pick" picking should have lost its partner and have its origin updated
-        move2._action_confirm()
+        # second out move, the "pick" picking should have lost its partner and have its origin updated.
```

Comment:
keep the diff minimal if you don't change a line don't change the history
```suggestion
        # second out move, the "pick" picking should have lost its partner and have its origin updated
```